dash-renderer/src/actions/setAppReadyState.js

```diff
@@ -1,87 +1,43 @@
-import {filter} from 'ramda';
-import {createAction} from 'redux-actions';
+import { path } from 'ramda';
+import { createAction } from 'redux-actions';
 
-import isSimpleComponent from '../isSimpleComponent';
 import Registry from './../registry';
-import {getAction} from './constants';
-import {isReady} from '@plotly/dash-component-plugins';
+import { getAction } from './constants';
+import { isReady } from '@plotly/dash-component-plugins';
 
-const isAppReady = layout => {
-    const queue = [layout];
-
-    const components = {};
-    const ids = {};
-
-    /* Would be much simpler if the Registry was aware of what it contained... */
-    while (queue.length) {
-        const elementLayout = queue.shift();
-        if (!elementLayout) {
-            continue;
-        }
-
-        const id = elementLayout.props && elementLayout.props.id;
-        const children = elementLayout.props && elementLayout.props.children;
-        const namespace = elementLayout.namespace;
-        const type = elementLayout.type;
-
-        components[namespace] = components[namespace] || {};
-        components[namespace][type] = type;
+const isAppReady = (layout, paths) => targets => {
```

Comment:
As long as we get the order of operations right (ie always updating `isAppReady` after any update to `layout` or `paths`) this should be fine, but it would be more robust and simpler - without any cost anymore - to not put this in the state at all.

Reply:
ie to just make it a function `(layout, paths, targets) => {...}`

Reply:
exactly the problem making the tests fail, the store evals `IS_APP_READY` before `COMPUTE_PATH` and the first calls ends up not blocking